src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -71,40 +71,29 @@ public string TagName
         /// The HTML element's pre content.
         /// </summary>
         /// <remarks>Value is prepended to the <see cref="ITagHelper"/>'s final output.</remarks>
-        public string PreContent { get; set; }
+        public TagHelperContent PreContent { get { return _preContent; } }
```

Comment:
nit: let properties breath.  we normally only do the one-line thing with auto-properties.